master/internal/api_auth.go

```diff
@@ -19,7 +32,7 @@ func (a *apiServer) Login(
 		return nil, err
 	}
 
-	if !user.ValidatePassword(req.Password) {
+	if !user.ValidatePassword(replicateClientSideSaltAndHash(req.Password)) {
```

Comment:
I think we need a special treatment for an empty password if it's not taking a different codepath. I have to look at the rest of the code

Reply:
Ah yea, good catch 🤦 .. I even have that in the other version of this code (from SCIM password sync impl).

Reply:
It's still fine this way though, because if they're a "password-less" user, we just dont check their password (which all users without passwords are)